merlin/study/step.py

```diff
@@ -277,9 +266,29 @@ def _update_status_file(
             "restarts": self.restarts,
         }
 
+        # Add celery specific info
+        if task_server == "celery":
+            from merlin.celery import app  # pylint: disable=C0415
+
+            # If the tasks are always eager, this is a local run and we won't have workers running
+            if not app.conf.task_always_eager:
+                status_info[self.name]["task_queue"] = get_current_queue()
+
+                # Add the current worker to the workspace-specific status info
+                current_worker = get_current_worker()
+                if "workers" not in status_info[self.name][self.condensed_workspace]:
+                    status_info[self.name][self.condensed_workspace]["workers"] = [current_worker]
+                elif current_worker not in status_info[self.name][self.condensed_workspace]["workers"]:
+                    status_info[self.name][self.condensed_workspace]["workers"].append(current_worker)
+
+                # Add the current worker to the overall-step status info
+                if "workers" not in status_info[self.name]:
+                    status_info[self.name]["workers"] = [current_worker]
+                elif current_worker not in status_info[self.name]["workers"]:
+                    status_info[self.name]["workers"].append(current_worker)
+
         LOG.info(f"Writing status for {self.name} to '{status_filepath}...")
-        with open(status_filepath, "w") as status_file:
-            json.dump(status_info, status_file)
+        write_status(status_info, status_filepath, f"{self.workspace.value}/status.lock")
```

Comment:
I know we’re not catching errors in here, since we will want to know that something went wrong, but is there a way to do that without killing the workflow? An I/O system glitch eg that is transient could take down the workflow if not gracefully handled

Reply:
Maybe something like I mention in my previous comment where we just catch anything and everything that could go wrong. We could instead log it with a warning, which will likely cause SOFT FAILs.

Reply:
I think if there’s a way we can grab bad things that happen and log them as SOFT_FAILS it’ll go a long way towards resilience

Reply:
As a related note (probably for a different PR) if we could do that for individual tasks, especially those in chords so we can head off chord errors, a bunch of our “frozen workflow” issues would go away (they’d become failed tasks instead, which is way better)

Reply:
From a high level that sounds super do-able.

Reply:
I think I know how to do this (thanks Google Gemini). We can define an errback to the chord that gets called. 

See: https://docs.celeryq.dev/en/stable/userguide/canvas.html#chords

Reply:
They have an example that just prints the error that was called. We can also set task_allow_error_cb_on_chord_header to allow for tasks in the header to fail too

Reply:
I remember playing with this in the past and noticing that if a chord errback is called then that will fully stop the workflow as the typical callback (in our case the `chordfinisher` task) will never get called. When the typical callback is never given then the next chord in the chain never fires off so everything just stalls.

Maybe there's a way to trigger the next chord in the chain manually, in which case we could do that at the end of the chord errback.